tests/common/include/tests/common/path_correction.inl

```diff
@@ -0,0 +1,524 @@
+/** Detray library, part of the ACTS project (R&D line)
+ *
+ * (c) 2022 CERN for the benefit of the ACTS project
+ *
+ * Mozilla Public License Version 2.0
+ */
+
+// Project include(s).
+#include "detray/definitions/units.hpp"
+#include "detray/materials/predefined_materials.hpp"
+#include "detray/propagator/actor_chain.hpp"
+#include "detray/propagator/actors/parameter_resetter.hpp"
+#include "detray/propagator/actors/parameter_transporter.hpp"
+#include "detray/propagator/line_stepper.hpp"
+#include "detray/propagator/navigator.hpp"
+#include "detray/propagator/propagator.hpp"
+#include "detray/propagator/rk_stepper.hpp"
+#include "detray/tracks/tracks.hpp"
+#include "tests/common/tools/detector_metadata.hpp"
+
+// Covfie include(s).
+#include <covfie/core/field.hpp>
+#include <covfie/core/field_view.hpp>
+
+// Vecmem include(s).
+#include <vecmem/memory/host_memory_resource.hpp>
+
+// google-test include(s).
+#include <gtest/gtest.h>
+
+using namespace detray;
+
+struct surface_targeter : actor {
+
+    struct state {
+        scalar _path;
+        dindex _target_surface_index = dindex_invalid;
+    };
+
+    /// Enforces thepath limit on a stepper state
+    ///
+    /// @param abrt_state contains the path limit
+    /// @param prop_state state of the propagation
+    template <typename propagator_state_t>
+    DETRAY_HOST_DEVICE void operator()(state &actor_state,
+                                       propagator_state_t &prop_state) const {
+
+        prop_state._heartbeat = true;
+        auto &navigation = prop_state._navigation;
+        auto &stepping = prop_state._stepping;
+        navigation.set_full_trust();
+
+        scalar residual = actor_state._path - stepping.path_length();
+        stepping.set_constraint(residual);
```

Comment:
This only makes sense if ```residual``` is smaller than the distance to the next surface (I don't know if it is?)